src/AutoRest.CSharp/Common/Input/CodeModelConverter.cs

```diff
@@ -148,7 +148,7 @@ public List<InputParameter> CreateOperationParameters(IReadOnlyCollection<Reques
             BodyType: GetResponseBodyType(response),
             BodyMediaType: GetBodyFormat(response.HttpResponse.KnownMediaType),
             Headers: GetResponseHeaders(response.HttpResponse.Headers),
-            IsErrorResponse: false,
+            IsErrorResponse: response.HttpResponse.IntStatusCodes.Any(code => code >= 400),
```

Comment:
code model has a flag to indicate whether the response is an error response. We should not do this.